### DISCUSSION TITLE
bRPC 源码分析「一、协程设计」 | SF-Zhou's Blog

### DISCUSSION THREAD

**Discussion by SF-Zhou** [SF-Zhou/sf-zhou.github.io#160] [Announcements]
https://sf-zhou.github.io/brpc/brpc_01_bthread.html

**Comment by four1er** [comment]
写的很好 ps：您这个博客用的主题是啥？很好看！

**Comment by SF-Zhou** [comment]
> @four1er
> 写的很好 ps：您这个博客用的主题是啥？很好看！

谢谢～ 博客是我自己设计的，用 Node.js 生成的全静态网页，源码可以在 GitHub 上找到。

**Comment by dandyhuang** [comment]
哪些经验可以用在工作上呀，大佬

**Reply by SF-Zhou** [reply]
bRPC 的单连接设计、wait-free 的写队列、内存池、协程调度等，好的设计都可以用在自己的工作上。比如我后面自己做 RPC 框架时，借鉴 bRPC 设计的[单 Epoll 多线程 IO 模型](https://sf-zhou.github.io/linux/single_epoll_multi_workers.html)。

**Reply by dandyhuang** [reply]
收到，谢谢大佬
